### PR TITLE
reloading crash & some small stuff

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] favorites would crash when reloading cause the dev part of the store isn't actually plumbed through
- [x] opening folders with ..(blah) would die due to not being surrounded in quote
- [x] added idle-callback to reduce cpu strain due to logging. seems to work better for me

@keybase/react-hackers

### CODE REVIEW

File: shared/actions/favorite.js [hunk 1]
```diff
@@ -198,7 +199,6 @@ export function favoriteList (): (dispatch: Dispatch, getState: () => Object) =>
 
         // figure out who can solve the rekey
         const folders: Array<FolderWithMeta> = jsonToFolders(json, myKID)
-        const config = getState && getState().config
```
Comment (chrisnojima) [inline shared/actions/favorite.js]: pull this up so we can reuse it

File: shared/menubar/index.js [hunk 2]
```diff
@@ -200,6 +200,9 @@ export function selector (): (store: Object) => Object {
         extendedConfig: store.config.extendedConfig,
       },
       favorite: store.favorite,
+      dev: {
```
Comment (chrisnojima) [inline shared/menubar/index.js]: dev wasn't being passed through

File: shared/util/idle-callback.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+// @flow
+const requestIdleCallback = typeof window !== 'undefined' && window.requestIdleCallback ||
```
Comment (chrisnojima) [inline shared/util/idle-callback.js]: suggested by google as a polyfill

File: shared/actions/favorite.js [hunk 4]
```diff
@@ -165,7 +165,7 @@ const jsonToFolders = (json: Object, myKID: any) => {
 export function favoriteList (): (dispatch: Dispatch, getState: () => Object) => void {
   return (dispatch, getState) => {
     // Ignore any messages we receive while reloading
-    if (getState().dev.reloading) {
+    if (getState().dev && getState().dev.reloading) {
```
Comment (MarcoPolo) [inline shared/actions/favorite.js]: If it was written like this originally, would you have found this bug as easily?

File: shared/store/action-logger.js [hunk 5]
```diff
@@ -1,12 +1,11 @@
 // @flow
-
-import {Iterable} from 'immutable'
 import deep from 'deep-diff'
+import type {StatSink} from '../util/stats'
```
Comment (MarcoPolo) [inline shared/store/action-logger.js]: 🐑 I like the pattern of having types separate from other imports

File: shared/store/action-logger.js [hunk 6]
```diff
@@ -39,9 +38,7 @@ const actionStatSink: StatSink = {
 }
 
 export const actionLogger = (store: any) => (next: any) => (action: any) => {
-  console.groupCollapsed && console.groupCollapsed(`Dispatching action: ${action.type}`)
-
-  console.log(`Dispatching action: ${action.type}: `, forwardLogs ? JSON.stringify(action) : action)
+  const log1 = `Dispatching action: ${action.type}: ${forwardLogs ? JSON.stringify(action) : action}`
```
Comment (MarcoPolo) [inline shared/store/action-logger.js]: This won't work. It'll become `[Object object]` (when not stringified). this should be an array that gets applied later.

File: shared/store/action-logger.js [hunk 7]
```diff
@@ -56,13 +53,19 @@ export const actionLogger = (store: any) => (next: any) => (action: any) => {
 
   startTiming(shouldRunLogStats, loggingStatSink)
   const diff = deep.diff(objToJS(oldState), objToJS(newState))
-  console.log('Diff:', forwardLogs ? JSON.stringify(diff) : diff)
+  const log2 = `Diff: ${forwardLogs ? JSON.stringify(diff) : diff}`
   endTiming(shouldRunLogStats, loggingStatSink)
 
-  console.groupEnd && console.groupEnd()
+  requestIdleCallback(() => {
```
Comment (MarcoPolo) [inline shared/store/action-logger.js]: are these guaranteed to queue up properly?

Comment (chrisnojima) [inline shared/store/action-logger.js]: if we don't add the optional timeout, which we don't

File: shared/actions/favorite.js (hunk 4 above)
Comment (chrisnojima) [inline shared/actions/favorite.js]: we don't want a crash having to do with this dev functionality, this is just for HMR so...

File: shared/store/action-logger.js (hunk 5 above)
Comment (chrisnojima) [inline shared/store/action-logger.js]: that'll have to wait till we get the linting stuff better

Comment (MarcoPolo) [inline shared/store/action-logger.js]: You don't have to change it, but I don't understand, it was already separated by a blank line, so you just had to sort the top imports (and the bottom type imports) individually. No need to move the type imports up and then sort.

**Comment by chrisnojima** [comment]
@MarcoPolo  updated

**Comment by MarcoPolo** [comment]
👍

**Pull request merged**